File: BucketList/app/src/main/java/com/example/bucketlist/MainActivity.java
package com.example.bucketlist;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.cardview.widget.CardView;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(android.R.id.content), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        // Set up card click listeners
        CardView cardThingsToDo = findViewById(R.id.card_things_to_do);
        CardView cardPlacesToGo = findViewById(R.id.card_places_to_go);

        cardThingsToDo.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, ThingsActivity.class);
                startActivity(intent);
            }
        });

        cardPlacesToGo.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this, PlacesActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: RecordKeeper/app/src/main/java/com/example/recordkeeper/MainActivity.kt
package com.example.recordkeeper

import android.os.Bundle
import android.view.Menu
import android.view.MenuInflater
import android.view.MenuItem
import android.view.View
import android.widget.FrameLayout
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.fragment.app.commit
import com.example.recordkeeper.databinding.ActivityMainBinding
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.google.android.material.navigation.NavigationBarView.OnItemSelectedListener

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)


//        supportFragmentManager.commit {
//            add(R.id.frame_content, RunningFragment())
//        }

        // Set up BottomNavigationView with proper listener
        binding.bottomNav.setOnItemSelectedListener { menuItem ->
            when (menuItem.itemId) {
                R.id.nav_running -> onRunningClicked()
                R.id.nav_cycling -> onCyclingClicked()
                else -> false
            }
        }



//        binding.bottomNav.setOnItemSelectedListener(this)

//        binding.btnCycling.setOnClickListener(object : View.OnClickListener{
//
//            override fun onClick(v: View?) {
//
//            }
//
//        } ) --> anonymous class

//        binding.btnCycling.setOnClickListener{
//           onCyclingClicked()
//        }
//
//        binding.btnRunning.setOnClickListener{
//           onRunningClicked()
//        }

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.toolbar, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle item selection.
        return when (item.itemId) {
            R.id.reset_running -> {
                Toast.makeText(this,"Reset Running menu item",Toast.LENGTH_SHORT).show()
                true
            }
            R.id.reset_cycling -> {
                Toast.makeText(this,"Reset Cycling menu item",Toast.LENGTH_SHORT).show()
                true
            }
            R.id.reset_all -> {
                Toast.makeText(this,"Reset All menu item",Toast.LENGTH_SHORT).show()
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    private fun onRunningClicked(): Boolean {
        supportFragmentManager.commit {
            replace(R.id.frame_content, RunningFragment())
        }
        return true
    }

    private fun onCyclingClicked(): Boolean {
        supportFragmentManager.commit {
            replace(R.id.frame_content, CyclingFragment())
        }
        return true
    }

//    override fun onClick(v: View?) {
//
//        Toast.makeText(this,"I have made the Activity a click listener",Toast.LENGTH_LONG).show()
//    }

}